### ISSUE TITLE
Wakeword problem on rpi 3 with Stretch

### ISSUE THREAD

**Issue by gbCambridge** [alexa/alexa-avs-sample-app#1042]
Hi, 
I have installed a DAC Hat board on my rpi. This has not interfered with the Alexa implementation....the voice still comes from the 3.5mm jack on the rpi pcb.
However, when I run wakeword  with sudo ./wakeWordAgent -e kitt_ai I get the following:

INFO:main: Starting Wake Word Agent
INFO:WakeWordAgent: State set to IDLE(2)
ALSA lib confmisc.c:1281:(snd_func_refer) Unable to find definition 'cards.bcm2835.pcm.front.0:CARD=0'
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5007:(snd_config_expand) Evaluate error: No such file or directory

etc..........................................................

When I check the Playback devices I get the following:
**** List of PLAYBACK Hardware Devices ****
card 0: ALSA [bcm2835 ALSA], device 0: bcm2835 ALSA [bcm2835 ALSA]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 0: ALSA [bcm2835 ALSA], device 1: bcm2835 ALSA [bcm2835 IEC958/HDMI]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ HiFi pcm512x-hifi-0 []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

As you can see the device that Wakeword is seeking is there, but not identified in a manner that is accepted by Wakeword. Alexa itself is happy to use these settings.  So far I have not been able to identify the culprit that zeroes the devices, but in the short term, I am wondering how best to get over this problem.
Alternatives seem to be ......1) Fix the device descriptions,  2) Fix wordwake to be more compatible with the way Alexa (apparently) works......Alexa is unaffected by these settings  3)...here I await your suggestions :-)

Any thoughts on the best way forward gratefully received

George

**Comment by scotthea-amazon** [comment]
Hello George,

On this page [http://docs.kitt.ai/snowboy](http://docs.kitt.ai/snowboy) they describe how to test if your audio routing is working on rpi and some suggestions for editing your configuration.

Also, there have been a number of people working through similar problems who have posted to this issue: #277.  You may find some useful hints there.

Regards,
-SWH

**Comment by shivasiddharth** [comment]
Hello George,
Can you check your asoundrc ? Sometimes, it gets auto populated by the OS with hw:0,0.

**Comment by gbCambridge** [comment]
Hi All,
Thanks for all the suggestions....I have checked all my files according to your recommendations....and no luck.  However, I came across the recommendation of trying the installation at https://github.com/alexa-pi/AlexaPi   and I can report that it is simple, does not need three terminal windows. It just runs...exactly as it says...and no funny business. So I have accepted defeat, and taken a different path... and it works very smoothly.
Once again, thanks for all the suggestions....it has really taken me back to Unix of old ;-)
